### ISSUE TITLE
Initial value for deeply nested forms when adding a field.

### ISSUE THREAD

**Issue by tokenvolt** [erikras/redux-form#377]
Is there any way I could add an initial value for nested form but only when I want to add it?
````js
export const fields = [
  "content",
  "children[].firstname",
  "children[].lastname"
];
````
And when I call `children.addField()` I want firstname to have an initial value. Looking into the code I see that addField receives 2 optional arguments and one of them is `value`, but if I try to use it

````js
children.addField({firstname: 'Tom', lastname: 'Whatever'})
````

it doesn't seem to work. I'm not sure I'm doing this right. 
I'm using v3.1.7.

**Comment by davidkpiano** [comment]
Doesn't work when passing in a key, either.

**Comment by Strajk** [comment]
Hey @tokenvolt, have you found some solution?

**Comment by davidkpiano** [comment]
@Strajk @tokenvolt the only real (ugly) solution right now is to do it field by field.

@erikras Would you be open to a helper method for this, like `field.addFields({firstname: 'Tom', lastname: 'Whatever'})` that iterates through each key/value pair and updates the field?

**Comment by erikras** [comment]
The initialization stuff is currently being rewritten to address #360 and all the issues linked to it. I'll keep this in mind as well.

**Comment by erikras** [comment]
This should be fixed in [`v4.0.0`](https://github.com/erikras/redux-form/releases/tag/v4.0.0). Come and get it!

**Comment by davecoates** [comment]
Would it be reasonable to handle populating multiple fields as in the OP example, ie

```
children.addField({firstname: 'Tom', lastname: 'Whatever'})
```

Currently this doesn't work because in the reducer it just does 

```
const newValue = {value};
```

Would it be reasonable to check if value is an object, and if so treat it as map of fields to values or could value legitimately be an object in other cases too? (I'm not sure, can a single field have an object as a field or is it always a scalar?)

eg

```
let newValue;
if (isObject(value)) {
   newValue = {}
   Object.keys(value).forEach(fieldName => newValue[fieldName] = {value: value[fieldName]})
} else {
   newValue = {value}
}
```
